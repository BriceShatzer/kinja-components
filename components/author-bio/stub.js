// @flow

import User from 'kinja-magma/models/User';

const authorUserProperties = {
	'1484116754': {
		showBio: true,
		twitterHandle: 'zmetser',
		pgpKeyUrl: 'https://keybase.io/zmetser',
		pgpFingerprint: 'CE505548D52848B1',
		otrFingerprint: 'CE505548D52848B1',
		bio: 'Hacking Kinja',
		email: 'test@kinja.com',
		emailIsConfirmed: true
	}
};

const authors = [
	{
		id: '1484116754',
		avatar: {
			id: '17jcxldy9dbzapng',
			format: 'jpg'
		},
		displayName: 'John Doe',
		screenName: 'joeKinja',
		isSuperuser: false,
		status: 'enabled',
		createdMillis: 0

	},
	{
		id: '1484116754',
		avatar: {
			id: '17jcxldy9dbzapng',
			format: 'jpg'
		},
		displayName: 'Sam Doe',
		screenName: 'samKinja',
		isSuperuser: false,
		status: 'enabled',
		createdMillis: 0
	}
];


export const props = {
	host: 'kinja.com',
	authors: authors.map<User>(author => User.fromJSON(author)),
	isAmp: false,
	authorUserProperties
};
