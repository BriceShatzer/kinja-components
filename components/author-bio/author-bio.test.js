/* @flow */

import * as React from 'react';
import { shallow } from 'enzyme';

import AuthorBio from './author-bio';
import { props } from './stub.js';

describe('<AuthorBio />', () => {
	it('should render AuthorBio by default', () => {
		const wrapper = shallow(<AuthorBio {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});
