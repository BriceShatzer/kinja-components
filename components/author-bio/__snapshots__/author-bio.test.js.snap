// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorBio /> should render AuthorBio by default 1`] = `
<EnsureDefaultTheme>
  <author-bio-style__AuthorWrapper>
    <Author
      authName={null}
      avatar={
        Object {
          "format": "jpg",
          "frozenFormat": "jpg",
          "id": "17jcxldy9dbzapng",
          "isAnimated": false,
        }
      }
      confirmedEmail={null}
      createdMillis={0}
      displayName="John Doe"
      host="kinja.com"
      id="1484116754"
      isEditStaff={false}
      isGmgMember={false}
      isSales={false}
      isSuperuser={false}
      key="17jcxldy9dbzapngjoeKinja"
      membership={Array []}
      remotes={Object {}}
      screenName="joeKinja"
      status="enabled"
      token={null}
      userProperties={
        Object {
          "bio": "Hacking Kinja",
          "email": "test@kinja.com",
          "emailIsConfirmed": true,
          "otrFingerprint": "CE505548D52848B1",
          "pgpFingerprint": "CE505548D52848B1",
          "pgpKeyUrl": "https://keybase.io/zmetser",
          "showBio": true,
          "twitterHandle": "zmetser",
        }
      }
    />
    <Author
      authName={null}
      avatar={
        Object {
          "format": "jpg",
          "frozenFormat": "jpg",
          "id": "17jcxldy9dbzapng",
          "isAnimated": false,
        }
      }
      confirmedEmail={null}
      createdMillis={0}
      displayName="Sam Doe"
      host="kinja.com"
      id="1484116754"
      isEditStaff={false}
      isGmgMember={false}
      isSales={false}
      isSuperuser={false}
      key="17jcxldy9dbzapngsamKinja"
      membership={Array []}
      remotes={Object {}}
      screenName="samKinja"
      status="enabled"
      token={null}
      userProperties={
        Object {
          "bio": "Hacking Kinja",
          "email": "test@kinja.com",
          "emailIsConfirmed": true,
          "otrFingerprint": "CE505548D52848B1",
          "pgpFingerprint": "CE505548D52848B1",
          "pgpKeyUrl": "https://keybase.io/zmetser",
          "showBio": true,
          "twitterHandle": "zmetser",
        }
      }
    />
  </author-bio-style__AuthorWrapper>
</EnsureDefaultTheme>
`;
